src/services/game/generateQuestions.js

```diff
@@ -0,0 +1,78 @@
+import { fetchNameFromAPI, fetchImgFromAPI } from '../swapi/swapi';
+
+const getRandomNumber = (minInt, maxInt) => {
+  return Math.floor(Math.random() * (maxInt - minInt + 1)) + minInt;
+};
+
+const getRangeOfIds = (mode) => {
```

Comment:
You are right that hardcoding is not really good practice and it will be better to fetch the ids and map them into an array. However, I run on the issue that this API is paginated and you can fetch an array of only 10 elements at once. E.g. if you try to fetch `const url = `${swapiEnv}/${gameMode}`;` this will result with:

`{count: 36, next: 'https://swapi.dev/api/starships/?page=2', previous: null, results: Array(10)}`

I looked up API pagination on google and found this topic on Stackoverflow from 2019 with the exact same question I have: [how to get all characters at once?](https://stackoverflow.com/questions/68848755/how-to-get-data-from-paginated-rest-api-using-webclient). The answer points to Django's API library and I cannot really find a solution to this issue. 

Another issue is that original API elements do not have any ids, the only data that contains their ids is url `url: "https://swapi.dev/api/starships/2/"`. So I would have to extract it from here.

While writing this answer I just came up with maybe a possible solution for this problem. The json files and data inside it have ids (`"pk": 6`) let me know if it would be possible to fetch data from local files. That would make it possible to fetch array with elements and map it to correct ids.

Reply:
It's quite strange that it automatically set query parameter of page as 2. 
But I think we don't have to use SWAPI API directly, we have implemented our server in repo.
You have to open project in 2nd console -> go to `swapi-json-server` -> npm install -> npm start.
Then instead of `process.env.SW_API_BASE_URL` use those value from `env.jsonserver.local` or simply put `http://localhost:3000`.